File: ai_tnhn/ai-api-tnhn/router/contract.go
package router

import (
	"ai-api-tnhn/handler"
	"ai-api-tnhn/router/middleware"

	"github.com/gin-gonic/gin"
)

func (h *HandlerFuncs) ContractRoutes(r *gin.RouterGroup, mid middleware.Middleware, handler *handler.ContractHandler) {
	group := r.Group("/contracts")
	// Accessible by Super Admin and Manager
	group.Use(mid.MidBasicType())
	{
		group.GET("", handler.List)
		group.GET("/:id", handler.GetByID)
		group.POST("/prepare-folder", handler.PrepareFolder)
		group.POST("", handler.Create)
		group.PUT("/:id", handler.Update)
		group.DELETE("/:id", handler.Delete)
		group.POST("/:id/upload", handler.Upload)
		group.POST("/upload-to-folder", handler.UploadToFolder)
		group.DELETE("/delete-file", handler.DeleteFile)
	}
}

File: ai_tnhn/ai-frontend-tnhn/src/api/contract.js
import axiosClient from './axiosClient';

const contractApi = {
    getAll: (params) => axiosClient.get('/admin/contracts', { params }),
    getById: (id) => axiosClient.get(`/admin/contracts/${id}`),
    create: (data) => axiosClient.post('/admin/contracts', data),
    update: (id, data) => axiosClient.put(`/admin/contracts/${id}`, data),
    delete: (id) => axiosClient.delete(`/admin/contracts/${id}`),
    upload: (id, formData) => axiosClient.post(`/admin/contracts/${id}/upload`, formData, {
        headers: { 'Content-Type': 'multipart/form-data' }
    }),
    prepareFolder: (data) => axiosClient.post('/admin/contracts/prepare-folder', data),
    uploadToFolder: (folderId, formData) => axiosClient.post(`/admin/contracts/upload-to-folder?folder_id=${folderId}`, formData, {
        headers: { 'Content-Type': 'multipart/form-data' }
    }),
    chatContract: (data) => axiosClient.post('/admin/google/contract-chat', data)
};

export default contractApi;
